### PR TITLE
Server tests

### PR DESCRIPTION
**Pull request by vogdb**
Should be done after #89.

### CODE REVIEW

**Comment by vogdb** [comment]
@tomaszdurka please review

**Comment by vogdb** [comment]
@tomaszdurka please review

**Comment by vogdb** [comment]
@tomaszdurka please look

**Comment by vogdb** [comment]
@tomaszdurka please review

File: lib/server.js [hunk 1]
```diff
@@ -33,11 +33,21 @@ module.exports = (function() {
    * @constructor
    */
   function Server(redisHost, statusPort) {
-    connectRedisDown(redisHost);
-    connectRedisUp(redisHost);
-    createStatusServer(statusPort);
+    /** @type {RedisClient} */
+    this._redisClientDown = connectRedisDown(redisHost);
```
Comment (tomaszdurka) [inline lib/server.js]: I'm ok with those changes, just am not huge fan of connect methods returning a client.
Let's maybe rename them to `createClientDown`?

**Comment by vogdb** [comment]
@tomaszdurka please review

**Comment by tomaszdurka** [comment]
lgtm

**Pull request merged**